### DISCUSSION TITLE
About graalVM JIT

### DISCUSSION THREAD

**Discussion by PyAntony** [quarkusio/quarkus#30911] [Q&A] [answered]
Hi,

According to my understanding Quarkus has nothing to do with GraalVM JIT isn't? Quarkus is ONLY concerned with native images through Mandrel (which have nothing to do with JIT compilation). If I want to use GraalVM JIT I have to install Graal and run my program within the Graal environment correct? (regular use of Graal according to it's documentation)

[a comment by an automated account is not shown]

**Comment by Karm** [comment, marked as answer]
@PyAntony 

> According to my understanding Quarkus has nothing to do with GraalVM JIT isn't?

Quarkus does not do anything to explicitly aid [GraalVM JIT compiler](https://www.graalvm.org/latest/reference-manual/java/compiler/) per se.

> Quarkus is ONLY concerned with native images through Mandrel (which have nothing to do with JIT compilation).

That is false. Quarkus can run with your JVM of choice, e.g. with HotSpot in OpenJDK. It could also be Ahead Of Time compiled into a single executable, utilizing Substrate VM as its runtime VM. You need a `native-image` tool for that. One can get that either from GraalVM distribution or from a Mandrel distribution. The diference between what you download as GraalVM and what you download as Mandrel is [outlined over here](https://github.com/graalvm/mandrel#how-does-mandrel-differ-from-graal).

> If I want to use GraalVM JIT I have to install Graal and run my program within the Graal environment correct? (regular use of Graal according to it's documentation)

That is correct. See [the docs](https://www.graalvm.org/latest/reference-manual/java/options/) on available options. Note that Mandrel's `java` command does not offer any of these Graal JIT options as Mandrel is actually a vanilla OpenJDK distribution **plus** native-image for Java ahead of time compilation. TL;DR: If you want to use GraalVM JIT, download GraalVM.

**Reply by PyAntony** [reply]
thanks!
